controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1133,6 +1136,39 @@ private boolean isStreamCutValidInternal(Map<Long, Long> streamCut, int epochLow
                     }
                 });
     }
+
+    @Override
+    public CompletableFuture<SimpleEntry<Long, Long>> getSplitMergeCountsTillEpoch(EpochRecord epochRecord) {
+        if (epochRecord.hasSplitMergeCounts()) {
+            return CompletableFuture.completedFuture(new SimpleEntry<>(epochRecord.getSplits(), epochRecord.getMerges()));
+        } else {
+            //migration case: build the cumulative count from all previous epochs
+            return fetchEpochs(0, epochRecord.getEpoch(), true).thenApply(this::mapToScaleMetadata)
```

Comment:
even in this case, i was thinking we can actually do better -- we dont need to create each of the epochs. fetchEpochs actually computes epochs from history timeseries. 
what we need is the history timeseries only. and doing one pass on history timeseries will enable us to get the splits and merges count. so we dont actually have to do the fetchEpochs and then maptoscalemetadata here. which will even avoid lots of objects being created in memory. It will be O(number of segments) memory that we will use if we just travese history time series as opposed to working with epoch records.

Reply:
we dont need to do this now.. i have created #5885 that we can work on for 0.10.0 and it will also help with streamcut related operations too.

Reply:
Ok. got it.